docs/architecture/ingest.md

```diff
@@ -0,0 +1,87 @@
+# Ingest
+
+This is a component that is responsible for ingesting and processing data.
+
+It exposes a GRPC API for Osmosis node clients to send data to be ingested.
+
+See [this ADR](https://www.notion.so/osmosiszone/ADR-006-Streaming-GRPC-SQS-Ingest-4e3b2ff7d23e43e2a1f3c43adc3c26bc) for details about the design.
+
+The node pushes only the pool data that is changed within a block.
+
+As a result, it is possible to see the following sequence of events:
+- Height X: All Osmosis pools are pushed
+- Height X+1: Only the pools that have changed within that height are pushed
+
+## Workers
+
+### Pricing
+
+The pricing worker is responsible for pre-computing prices for all assets and the default quote denom (USDC).
+
+It is triggerred by the ingester asyncronously when the GRPC push is received.
+
+It looks at the unique denoms constructed from the pool data and computes the prices for each
+asset and the default quote denom.
+
+Once complete, it calls a hook to notify the subscribed listeners that the prices have been updated.
+
+#### Pricing Listeners
+
+- Healthcheck: The healthcheck listener is responsible for updating the healthcheck status based on the last time the prices were updated. If the prices are not updated within a certain time period, the healthcheck status will be updated to unhealthy.
+An example of a listener is the healtcheck component. Once it is notified of the price update, it
```

Comment:
Sentence not finished after "Once it is notified of the price update, it" ?

Reply:
removed it